copy/compression.go

```diff
@@ -157,7 +157,7 @@ func (ic *imageCopier) bpcCompressUncompressed(stream *sourceStream, detected bp
 // bpcRecompressCompressed checks if we should be recompressing a compressed input to another format, and returns a *bpCompressionStepData if so.
 func (ic *imageCopier) bpcRecompressCompressed(stream *sourceStream, detected bpDetectCompressionStepData) (*bpCompressionStepData, error) {
 	if ic.c.dest.DesiredLayerCompression() == types.Compress && detected.isCompressed &&
-		ic.compressionFormat != nil && ic.compressionFormat.Name() != detected.format.Name() {
+		ic.compressionFormat != nil && ic.compressionFormat.InternalUnstableUndocumentedMIMEQuestionMark() != detected.format.InternalUnstableUndocumentedMIMEQuestionMark() {
```

Comment:
If we have a non-chunked zstd, and the user asks for chunked, shouldn’t we recompress?

Similarly if there is a non-chunked zstd on a registry, and the user asks for chunked push, shouldn’t we re-compress and re-push?

So at a quick glance this needs a more accurate modeling of “is at least as capable as”, or something.

Reply:
if you just inspect the tar blob, it is a regular zstd file.

Not sure how we can treat "zstd:chunked" as a completely separated algorithm.  I'll drop this patch for now and try something else

Reply:
Guessing, maybe the `blobPipelineDetectCompressionStep` could (itself, or preferably delegating to `pkg/compression`) combine the blob inspection with data from `srcInfo.Annotations`.

Reply:
what do you think of the approach here?

- https://github.com/containers/storage/pull/1756
- https://github.com/containers/image/pull/2183

Reply:
See in there; in short, AFAICS we _must_ have the annotation values for the chunked format to be consumable.

Reply:
This was removed from this PR. I vaguely remember we ended up with #2185 instead, for now.